File: 6-kyu/ip-validation/python/solution.py
import string

def is_valid_IP(strng):
    if not strng or not (strng.count('.') == 3):
        return False
    for n in strng.split('.'):
        if (n[0] == '0' and len(n) > 1) or (not n.isdigit()) or (int(n) > 255 or int(n) < 0):
            return False
    return True

File: 6-kyu/ip-validation/javascript/solution.js
const isValidIP = str => {
  const octets = str.split(".");
  if (octets.length !== 4) {
    return false;
  }
  if (!octets.every(oct => /^\d{1,3}$/g.test(oct)) || octets.some(oct => oct.length !== 1 && oct[0] === "0")) {
    return false;
  }
  const octetsNum = octets.map(Number);
  if (octetsNum.some(n => n < 0 || n > 255)) {
    return false;
  }
  return true;
};